optaplanner-core/src/main/java/org/optaplanner/core/api/score/stream/ConstraintCollectors.java

```diff
@@ -69,6 +69,21 @@
  */
 public final class ConstraintCollectors {
 
+    public static <A> DefaultUniConstraintCollector<A, ?, Object> loadBalanceByCount() {
+        return new DefaultUniConstraintCollector<>(
+                () -> new long[2],
+                (resultContainer, a) -> {
+                    resultContainer[0]++;
+                    long count = resultContainer[0];
+                    resultContainer[1] += (2 * count - 1);
+                    return () -> {
+                        resultContainer[1] -= (2 * resultContainer[0] - 1);
+                        resultContainer[0]--;
+                    };
+                },
+                resultContainer -> resultContainer);
+    }
```

Comment:
I am not sure if I am comfortable putting this into the API directly. I wouldn't mind having it in a TennisCollector first to let it ripe a bit :)

Reply:
Especially because it doesn't combine well with other constraints on the same level.

Reply:
Agree. This is just a rebase of an old PR, on which you made the same comment. :-)
I will work on it next week.

Reply:
Ah lol - happy to hear that I am consistent (this time at least) with my old self :)